go/fn/errors.go

```diff
@@ -38,10 +38,14 @@ func (e *ErrOpOrDie) Error() string {
 		e.obj.GetAPIVersion(), e.obj.GetKind(), e.obj.GetName(), strings.Join(e.fields, "/"))
 }
 
-func handleOptOrDieErr() {
+func handlePanic() {
 	if v := recover(); v != nil {
 		if e, ok := v.(ErrOpOrDie); ok {
 			log.Fatalf(e.Error())
+		} else if e, ok := v.(Result); ok {
+			log.Fatalf(e.Error())
```

Comment:
I guess we don't want it to exit here.

Reply:
I changed it to log the error to `Results` and return the output and error, but to do that I inlined the recovery function into `Run`. LMK if there is a better approach.